test/net/http/test_http_request.rb

```diff
@@ -89,5 +89,45 @@ def test_header_set
                          'Bug #7831 - do not decode content if the user overrides'
   end if Net::HTTP::HAVE_ZLIB
 
+  def test_deconstruct_keys
+    uri = URI('https://example.com/api/users')
+    req = Net::HTTP::Post.new(uri)
+    req.body = 'test data'
+    req['content-type'] = 'application/json'
+
+    keys = req.deconstruct_keys(nil)
+    assert_equal 'POST', keys[:method]
+    assert_equal '/api/users', keys[:path]
+    assert_equal uri, keys[:uri]
+    assert_equal 'test data', keys[:body]
+    assert_equal 'application/json', keys[:content_type]
+  end
+
+  def test_deconstruct_keys_with_specific_keys
+    req = Net::HTTP::Get.new('/test')
+
+    keys = req.deconstruct_keys([:method, :path])
+    assert_equal({method: 'GET', path: '/test'}, keys)
+  end
+
+  def test_pattern_matching
+    uri = URI('https://example.com/api/users')
+    req = Net::HTTP::Post.new(uri)
+
+    begin
+      matched = instance_eval <<~RUBY, __FILE__, __LINE__ + 1
```

Comment:
This was a similar trick as to what we had used around pattern matching for the 2.6 to 2.7 push, so I'm repeating it here. The other methods should function even without syntax bridges like this.